File: app/models/bet.rb
class Bet < ApplicationRecord
  monetize :price_cents

  belongs_to :game
  belongs_to :user

  validates :market, presence: true

  def teams
    game = Game.find(game_id) 

    "#{game.blue_side_team.name} vs #{game.red_side_team.name}"
  end

  def team_bet_on
    Team.find(team_bet_on_id).name
  end
end

File: app/javascript/src/types/bet.ts
import { Game } from './game'

export enum BetStatus {
  Won,
  Lost,
  AwaitingResult
}

export interface Bet {
  status: BetStatus
  id: number
  teamBetOnId: number
  market: string
  priceCents: number
  odds: number
  gameId: number
  game?: Game
  won?: boolean
}

export const setBetStatus = (won: boolean | null) => {
  if (won === null)
    return BetStatus.AwaitingResult

  return won ? BetStatus.Won : BetStatus.Lost
}
